tfx/components/example_gen/utils.py

```diff
@@ -262,31 +311,38 @@ def calculate_splits_fingerprint_and_span(
       Span number.
 
   Returns:
-    A Tuple of [fingerprint, select_span], where select_span is either
-    the value matched with the {SPAN} placeholder, or None if the placeholder
+    A Tuple of [fingerprint, select_span, select_version], where select_span
+    is either the value matched with the {SPAN} placeholder, or '0' if the
+    placeholder wasn't specified, and where select_version is either the
+    value matched with the {VERSION} placeholder, or '0' if the placeholder
```

Comment:
shall we distinguish a span = 0, vs. a span is None as before, similar to version.
we let the callsite of this function decide how to interpret None.

Reply:
agreed, i've updated the `calculate_splits_fp_span_and_version` to interprete what a None property means.